lighthouse-core/audits/first-meaningful-paint.js

```diff
@@ -120,27 +120,14 @@ class FirstMeaningfulPaint extends Audit {
   }
 
   /**
-   * @param {!Array<!Object>} traceData
+   * @param {!Object} tabTrace
    */
-  static collectEvents(traceData) {
-    // Parse the trace for our key events and sort them by timestamp.
-    const events = traceData.filter(e => {
-      return e.cat.includes('blink.user_timing') || e.name === 'TracingStartedInPage';
-    }).sort((event0, event1) => event0.ts - event1.ts);
-
-    // The first TracingStartedInPage in the trace is definitely our renderer thread of interest
-    // Beware: the tracingStartedInPage event can appear slightly after a navigationStart
-    const startedInPageEvt = events.find(e => e.name === 'TracingStartedInPage');
-    // Filter to just events matching the frame ID for sanity
-    const frameEvents = events.filter(e => e.args.frame === startedInPageEvt.args.data.page);
-
-    // Find our first FCP
-    const firstFCP = frameEvents.find(e => e.name === 'firstContentfulPaint');
-    // Our navStart will be the latest one before fCP.
-    const navigationStart = frameEvents.filter(e =>
-        e.name === 'navigationStart' && e.ts < firstFCP.ts).pop();
+  static collectEvents(tabTrace) {
+    const navigationStart = tabTrace.navigationStartEvt;
+    const firstFCP = tabTrace.firstContentfulPaintEvt;
+
     // fMP will follow at/after the FCP, though we allow some timestamp tolerance
-    const firstMeaningfulPaint = frameEvents.find(e =>
+    const firstMeaningfulPaint = tabTrace.traceEvents.find(e =>
```

Comment:
I know this is an FMP audit but seems a little strange that it's missing from the TraceOfTab while FCP is still there. Does FCP never fire twice but FMP does and that's why we have to worry about it here?